module/src/components/image/image.component.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { HTMLAttributes } from 'react';
+
+import { useElementStatus } from '../../hooks/useElementStatus';
+import { useHasTimeElapsedSinceMount } from '../../hooks/useHasTimeElapsed';
+import { IUseInViewportOptions, useInViewport } from '../../hooks/useIsInViewport';
+import { ClassNames } from '../../utils/classNames';
+import { IStatusProps, Status } from '../status';
+
+export interface IImageSource {
+  /** (string) the url source which will replace the main src if the right conditions are met */
+  srcSet: string;
+
+  /** (string) A media query used to decide whether to render this source I.E. "(min-width: 500px)" (see [picture on mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture)) */
+  media?: string;
+
+  /** (string) The type of the linked image source, I.E. "image/jpg", used by the browser to determine which image to show based on support of file types */
+  type?: string;
```

Comment:
Should this be 'mimeType' rather than 'type'?

Reply:
It's based on the `<source type="XXXX" />` element which uses the word `type` but you're right that it is a MIME type